packages/now-cli/src/util/env/get-env-records.ts

```diff
@@ -1,18 +1,62 @@
 import { Output } from '../output';
 import Client from '../client';
-import { ProjectEnvVariable, ProjectEnvTarget } from '../../types';
+import {
+  ProjectEnvVariable,
+  ProjectEnvTarget,
+  PaginationOptions,
+} from '../../types';
+
+type ApiVersion = 4 | 5;
+
+type APIV4Response = ProjectEnvVariable[];
+
+interface APIV5Response {
+  pagination: PaginationOptions;
+  envs: ProjectEnvVariable[];
+}
 
 export default async function getEnvVariables(
   output: Output,
   client: Client,
   projectId: string,
-  target?: ProjectEnvTarget
-): Promise<ProjectEnvVariable[]> {
+  apiVersion: 4,
+  target?: ProjectEnvTarget,
+  next?: number
+): Promise<APIV4Response>;
+
+export default async function getEnvVariables(
+  output: Output,
+  client: Client,
+  projectId: string,
+  apiVersion: 5,
+  target?: ProjectEnvTarget,
+  next?: number
+): Promise<APIV5Response>;
+
+export default async function getEnvVariables<V extends ApiVersion>(
+  output: Output,
+  client: Client,
+  projectId: string,
+  apiVersion: V,
+  target?: ProjectEnvTarget,
+  next?: number
+) {
   output.debug(
     `Fetching Environment Variables of project ${projectId} and target ${target}`
   );
   const qs = target ? `?target=${encodeURIComponent(target)}` : '';
-  const url = `/v4/projects/${projectId}/env${qs}`;
-  const records = await client.fetch<ProjectEnvVariable[]>(url);
-  return records;
+  const limit = apiVersion >= 5 ? '?limit=20' : '';
```

Comment:
There's a bug here when `target` is assigned and `apiVersion` is 5 because you'll end with two question marks.

Use use `new URL()` constructor instead and then the query string can be built properly with `searchParams` https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams